Add target labels

astorrs opened the pull request:
Add target label back in as optional, this will allow for a gradual migration.

astorrs committed:
Add target labels

Add target label back in as optional, this will allow for a gradual migration.

roreid commented:
@astorrs Would we also need to rollback or at least make the domain fields all optional for now? Wouldn't it cause validation errors for all existing schemas that don't have `domain.level1`?
![image](https://user-images.githubusercontent.com/66846698/179865980-f0177f2e-413a-4777-a69e-56029c599678.png)

astorrs committed:
Make data domain optional

Temporarily make the domains object as optional until we're ready to update the majority of schemas.

astorrs commented:
> @astorrs Would we also need to rollback or at least make the domain fields all optional for now? Wouldn't it cause validation errors for all existing schemas that don't have `domain.level1`? ![image](https://user-images.githubusercontent.com/66846698/179865980-f0177f2e-413a-4777-a69e-56029c599678.png)

@roreid Good call. I've updated to make domain optional.
